tests/integration/test_dbstatecheck.py

```diff
@@ -73,7 +73,7 @@ def test_basic(checker):
         ['output', '10000101T0000Z', 'succeeded'],
         ['output', '10010101T0000Z', 'succeeded'],
         ['good', '10000101T0000Z', 'waiting', '(flows=2)'],
-    ]
+        ['good', '10010101T0000Z', 'waiting', '(flows=2)'], ]
```

Comment:
Note, this task appears in the DB because the workflow has run on from the triggered task. This is an unintended side-effect of the `await sleep(1)` in the test fixture.

I've opened a PR to replace the sleep with a DB update call https://github.com/cylc/cylc-flow/pull/6212, however, if we want to keep this behaviour to allow the workflow to run to completion again (to test this PR), then replace the `sleep` with `await mod_complete(schd)` (and close #6212).

Reply:
(I've not had time to think about this today)

Reply:
(Reviewers - this is test framework issue, so not a reason to delay this PR)